Missing xtensa/config headers when building xtensa-intel_ace30_ptl_zephyr-elf tests

There are many build failures due to missing xtensa/config headers when building Zephyr tests and samples for `xtensa-intel_ace30_ptl_zephyr-elf`:

```
In file included from /__w/sdk-ng/sdk-ng/zephyrproject/zephyr/include/zephyr/arch/syscall.h:25,
                 from /__w/sdk-ng/sdk-ng/zephyrproject/zephyr/include/zephyr/syscall.h:12,
                 from /__w/sdk-ng/sdk-ng/test/twister-out/intel_adsp_ace30_ptl/tests/lib/c_lib/common/libraries.libc.common.picolibc.module/zephyr/include/generated/zephyr/syscalls/device.h:14,
                 from /__w/sdk-ng/sdk-ng/zephyrproject/zephyr/include/zephyr/device.h:1195,
                 from /__w/sdk-ng/sdk-ng/zephyrproject/zephyr/kernel/include/kernel_offsets.h:6,
                 from /__w/sdk-ng/sdk-ng/zephyrproject/zephyr/arch/xtensa/core/offsets/offsets.c:7:
/__w/sdk-ng/sdk-ng/zephyrproject/zephyr/include/zephyr/arch/xtensa/syscall.h:27:10: fatal error: xtensa/config/core-isa.h: No such file or directory
   27 | #include <xtensa/config/core-isa.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
ninja: build stopped: subcommand failed.
```

e.g. https://github.com/zephyrproject-rtos/sdk-ng/actions/runs/10925742016/job/30345709563#step:15:7906

--- a/west.yml
+++ b/west.yml
@@ -253,7 +253,7 @@ manifest:
       groups:
         - hal
     - name: hal_xtensa
-      revision: a2d658525b16c57bea8dd565f5bd5167e4b9f1ee
+      revision: baa56aa3e119b5aae43d16f9b2d2c8112e052871
       path: modules/hal/xtensa
       groups:
         - hal

